# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/mesa/hotfix-llvm-17.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

src/gallium/drivers/r600/sfn/sfn_valuefactory.cpp: In member function 'r600::Register* r600::ValueFactory::dest_from_string(const std::string&)':
src/gallium/drivers/r600/sfn/sfn_valuefactory.cpp:666:18: error: 'numeric_limits' is not a member of 'std'

--- mesa-23.1.7/src/gallium/drivers/r600/sfn/sfn_valuefactory.cpp.vanilla	2023-09-19 15:32:13.391413538 +0200
+++ mesa-23.1.7/src/gallium/drivers/r600/sfn/sfn_valuefactory.cpp	2023-09-19 15:32:56.723410689 +0200
@@ -34,6 +34,7 @@
 #include <iostream>
 #include <queue>
 #include <sstream>
+#include <limits>
 
 namespace r600 {
 
diff --unified --recursive --text mesa/src/gallium/auxiliary/gallivm/lp_bld_init.c mesa-new/src/gallium/auxiliary/gallivm/lp_bld_init.c
--- mesa/src/gallium/auxiliary/gallivm/lp_bld_init.c	2023-03-20 12:22:03.861856125 +0100
+++ mesa-new/src/gallium/auxiliary/gallivm/lp_bld_init.c	2023-03-20 13:07:34.218833163 +0100
@@ -42,8 +42,7 @@
 
 #include <llvm/Config/llvm-config.h>
 #include <llvm-c/Analysis.h>
-#include <llvm-c/Transforms/Scalar.h>
-#if LLVM_VERSION_MAJOR >= 7
+#if LLVM_VERSION_MAJOR >= 7 && LLVM_VERSION_MAJOR <17
 #include <llvm-c/Transforms/Utils.h>
 #endif
 #include <llvm-c/BitWriter.h>
